container-search/src/main/java/com/yahoo/prelude/fastsearch/VespaBackend.java

```diff
@@ -228,7 +227,11 @@ protected String ensureLegalSummaryClass(Query query, String summaryClass) {
             } else {
                 var db = getDocumentDatabase(query);
                 if (db != null && ! db.getDocsumDefinitionSet().hasDocsum(summaryClass)) {
-                    throw new IllegalInputException("invalid presentation.summary=" + summaryClass);
+                    if (summaryClass == query.getPresentation().getSummary()) {
+                        throw new IllegalInputException("invalid presentation.summary=" + summaryClass);
+                    } else {
+                        throw new IllegalArgumentException("invalid fill() with summaryClass=" + summaryClass);
```

Comment:
What's the difference of using `IllegalArgumentException` vs `IllegalInputException` in this context?